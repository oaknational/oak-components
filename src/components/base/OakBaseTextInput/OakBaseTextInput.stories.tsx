import React from "react";
import { Meta, StoryObj } from "@storybook/react";

import { OakBaseTextInput } from "./OakBaseTextInput";

/**
 *
 * An unstyled input to be used as a basis for UI input components.
 * The following callbacks are available for tracking focus events:
 *
 *  ### onFocus
 * `(e: FocusEvent<HTMLInputElement>) => void;`
 *  ### onBlur
 * `(e: FocusEvent<HTMLInputElement>) => void;`
 *  ### onInitialFocus
 * `(e: FocusEvent<HTMLInputElement>) => void;`<br>
 *  occurs only when the input is focused for the first time
 *
 */

const meta: Meta<typeof OakBaseTextInput> = {
  component: OakBaseTextInput,
  tags: ["autodocs"],
  title: "components/base/OakBaseTextInput",
  argTypes: {},
  parameters: {
    controls: {
      include: [],
    },
  },
};
export default meta;

type Story = StoryObj<typeof OakBaseTextInput>;

export const Default: Story = {
  render: (args) => (
    <OakBaseTextInput {...args} placeholder="placeholder text" type="text" />
  ),
  args: {},
  parameters: {
    controls: { include: [] },
  },
};
