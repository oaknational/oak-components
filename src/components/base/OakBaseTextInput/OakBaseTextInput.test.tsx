import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { create } from "react-test-renderer";

import { OakBaseTextInput } from "./OakBaseTextInput";

describe("OakBaseTextInput", () => {
  it("renders", () => {
    const { getByTestId } = render(<OakBaseTextInput data-testid="test" />);
    expect(getByTestId("test")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const tree = create(<OakBaseTextInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with placeholder", () => {
    const { getByPlaceholderText } = render(
      <OakBaseTextInput placeholder="placeholder text" />,
    );
    expect(getByPlaceholderText("placeholder text")).toBeInTheDocument();
  });

  it("calls onFocus when focused", () => {
    const onFocus = jest.fn();
    const { getByTestId } = render(
      <OakBaseTextInput data-testid="test" onFocus={onFocus} />,
    );
    getByTestId("test").focus();
    getByTestId("test").blur();
    getByTestId("test").focus();
    getByTestId("test").blur();
    getByTestId("test").focus();
    expect(onFocus).toHaveBeenCalledTimes(3);
  });

  it("calls onBlur when blurred", () => {
    const onBlur = jest.fn();
    const { getByTestId } = render(
      <OakBaseTextInput data-testid="test" onBlur={onBlur} />,
    );
    getByTestId("test").focus();
    getByTestId("test").blur();
    expect(onBlur).toHaveBeenCalled();
  });

  it("calls onInitialFocus when focused for the first time", () => {
    const onInitialFocus = jest.fn();
    const { getByTestId } = render(
      <OakBaseTextInput data-testid="test" onInitialFocus={onInitialFocus} />,
    );
    getByTestId("test").focus();
    expect(onInitialFocus).toHaveBeenCalled();
  });

  it("doesn't call onInitialFocus only when focused for subsequent times", () => {
    const onInitialFocus = jest.fn();
    const { getByTestId } = render(
      <OakBaseTextInput data-testid="test" onInitialFocus={onInitialFocus} />,
    );
    getByTestId("test").focus();
    getByTestId("test").blur();
    getByTestId("test").focus();
    getByTestId("test").blur();
    getByTestId("test").focus();
    expect(onInitialFocus).toHaveBeenCalledTimes(1);
  });
});
