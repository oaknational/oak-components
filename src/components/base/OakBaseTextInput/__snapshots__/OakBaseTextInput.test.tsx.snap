// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakBaseTextInput matches snapshot 1`] = `
<input
  className="sc-aXZVg dVeUxO"
  onBlur={[Function]}
  onFocus={[Function]}
/>
`;
